openmc/nuclide.py

```diff
@@ -93,3 +96,293 @@ def scattering(self, scattering):
+
+        if divisor_types:
+            cv.check_length('divisor types', divisor_types, len(types),
+                            len(types))
+            Ediv, data_div = self.calculate_xs(divisor_types, temperature,
+                                               sab_name, cross_sections)
+
+            # Create a new union grid, interpolate data and data_div on to that
+            # grid, and then do the actual division
+            Enum = E[:]
+            E = np.union1d(Enum, Ediv)
+            data_new = []
+
+            for l in range(len(types)):
+                data_new.append(openmc.data.Combination([data[l], data_div[l]],
+                                                        [np.divide]))
+                if divisor_types[l] != 'unity':
+                    types[l] = types[l] + ' / ' + divisor_types[l]
+            data = data_new
+
+        # Generate the plot
+        if axis is None:
+            fig = plt.figure(**kwargs)
+            ax = fig.add_subplot(111)
+        else:
+            fig = None
+            ax = axis
+        for i in range(len(data)):
+            to_plot = data[i](E)
+            # Set to loglog or semilogx depending on if we are plotting a data
+            # type which we expect to vary linearly
+            if types[i] in PLOT_TYPES_LINEAR:
+                plot_func = ax.semilogx
+            else:
+                plot_func = ax.loglog
+            if np.sum(to_plot) > 0.:
+                plot_func(E, to_plot, label=types[i])
+
+        ax.set_xlabel('Energy [eV]')
+        if divisor_types:
+            ax.set_ylabel('Microscopic Data')
+        else:
+            ax.set_ylabel('Microscopic Cross Section [b]')
+        ax.legend(loc='best')
+        ax.set_xlim(Erange)
+        if self.name is not None:
+            title = 'Microscopic Cross Section for ' + self.name
+            ax.set_title(title)
+
+        return fig
+
+    def calculate_xs(self, types, temperature=294., sab_name=None,
+                     cross_sections=None):
+        """Calculates continuous-energy cross sections of a requested type
+
+        Parameters
+        ----------
+        types : Iterable of str or Integral
+            The type of cross sections to calculate; values can either be those
+            in openmc.PLOT_TYPES or integers which correspond to reaction
+            channel (MT) numbers.
+        temperature : float, optional
+            Temperature in Kelvin to plot. If not specified, a default
+            temperature of 294K will be plotted. Note that the nearest
+            temperature in the library for each nuclide will be used as opposed
+            to using any interpolation.
+        sab_name : str, optional
+            Name of S(a,b) library to apply to MT=2 data when applicable.
+        cross_sections : str, optional
+            Location of cross_sections.xml file. Default is None.
+
+        Returns
+        -------
+        E : numpy.array
```

Comment:
Can you make this `energy_grid` to be consistent with the `element.calculate_xs()` method?